frontend/ui/uimanager.lua

```diff
@@ -1134,26 +1157,27 @@ function UIManager:_repaint()
     --]]
 
     for i = start_idx, #self._window_stack do
-        local widget = self._window_stack[i]
+        local window = self._window_stack[i]
+        local widget = window.widget
         -- paint if current widget or any widget underneath is dirty
-        if dirty or self._dirty[widget.widget] then
+        if dirty or self._dirty[widget] then
             -- pass hint to widget that we got when setting widget dirty
             -- the widget can use this to decide which parts should be refreshed
-            logger.dbg("painting widget:", widget.widget.name or widget.widget.id or tostring(widget))
+            logger.dbg("painting widget:", widget.name or widget.id or tostring(widget))
```

Comment:
The confusion between `window` and `widget` led to a very old and stupid bug, where, for anonymous widgets, this was logging the reference of the `window`, while both `show` & `close` log the reference of the `widget` (which was, obviously, always the intended behavior).